app/assets/javascripts/discourse/app/components/anonymous-topic-footer-buttons.js

```diff
@@ -1,21 +1,20 @@
 import Component from "@ember/component";
 import { computed } from "@ember/object";
+import { attributeBindings } from "@ember-decorators/component";
 import { getTopicFooterButtons } from "discourse/lib/register-topic-footer-button";
 
-export default Component.extend({
-  elementId: "topic-footer-buttons",
+@attributeBindings("role")
+export default class AnonymousTopicFooterButtons extends Component {
+  elementId = "topic-footer-buttons";
+  role = "region";
 
-  attributeBindings: ["role"],
-
-  role: "region",
-
-  allButtons: getTopicFooterButtons(),
+  @getTopicFooterButtons() allButtons;
```

Comment:
I think non-decorator version makes more sense here?

```suggestion
  allButtons = getTopicFooterButtons();
```